pkg/controller/controller_test.go

```diff
@@ -357,11 +350,12 @@ func TestControllerRun(t *testing.T) {
 		controller, _, _ := newWorkflowControllerFromClients(kubeClient, workflowClient)
 		controller.JobSynced = tc.JobSynced
 		controller.WorkflowSynced = tc.WorkflowSynced
+		expectedErrorMessage := tc.expectedErrorMessage
```

Comment:
Is this the data race fix that is mentionned in the PR description?  How does this change fix a race?  I'm missing the mechanics of the change.